src/core/algorithms/fd/hyfd/model/fd_tree_vertex.cpp

```diff
@@ -35,6 +37,8 @@ void FDTreeVertex::GetFdAndGeneralsRecursive(boost::dynamic_bitset<> const& lhs,
                                              std::vector<boost::dynamic_bitset<>>& result) const {
     if (IsFd(rhs)) {
         result.push_back(cur_lhs);
+        return;  // If this vertex is valid minimal FD, than all its specializations will be not
+                 // minimal FDs
```

Comment:
```suggestion
        return;  // If this vertex has the RHS bit set, then none of its children will have this bit set.
```
A vertex is not an FD.

Here is how I understand what's happening here.
Each vertex in this trie represents an FD's LHS. FD LHSs are partially ordered by inclusion (`X` <= `Y` iff `X` \subseteq `Y`).
We can order the vertices themselves on the LHSs they represent. This structure's invariant is then this: none of the vertices following the one with bit `n` set have the bit `n` also set.
In particular, the child vertices always follow the parent in that order (they represent the parent's LHSs with added elements, so the parent's LHS is a subset). That's the fact that is used here.